cpp/include/cudf/detail/gather.cuh

```diff
@@ -674,13 +674,20 @@ std::unique_ptr<table> gather(table_view const& source_table,
   }
 
   auto const nullable = bounds_policy == out_of_bounds_policy::NULLIFY ||
-                        std::any_of(source_table.begin(), source_table.end(), [](auto const& col) {
-                          return col.nullable();
-                        });
+                        cudf::has_nested_nullable_columns(source_table);
   if (nullable) {
-    auto const op = bounds_policy == out_of_bounds_policy::NULLIFY ? gather_bitmask_op::NULLIFY
-                                                                   : gather_bitmask_op::DONT_CHECK;
-    gather_bitmask(source_table, gather_map_begin, destination_columns, op, stream, mr);
+    auto const has_nulls =
+      bounds_policy == out_of_bounds_policy::NULLIFY || cudf::has_nested_nulls(source_table);
```

Comment:
Should this uses `&&` instead? Because if we indeed don't have any nulls here then we don't need to call `gather_bitmask`.

Reply:
Probably I misunderstood the usage of this variable. So this variable should be called `need_new_bitmask` or so. It should not be `has_nulls`.

Reply:
But we need to call `gather_bitmask` if `out_of_bounds_policy::NULLIFY`. `gather_bitmask` will help nullify any OOB accesses.

Reply:
Okay, so `||` is indeed needed, but please rename that variable.

Reply:
Done